BackendServer/Models/ServerMessage.cs

```diff
@@ -0,0 +1,27 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace BackendServer.Models
+{
+    enum MessageType
+    {
+        Info,
+        Warn,
+        Error
+    }
+
+    internal class ServerMessage(MessageType messageType, string content) : Message(content)
+    {
+        private MessageType messageType = messageType;
+        public MessageType MessageType
+        {
+            get { return messageType; }
+            set { messageType = value; }
+        }
+
+        public override string ParseRawMessage() { return Content; }
```

Comment:
ParseRawMessage is the function responsible for taking in a raw string and processing it into the current class member variables, shouldn't return anything?

Reply:
Changed ParseRawMessage() to protected void

Reply:
Having ParseRawMessage() return void and also be abstract means that the compiler can't know that we will fill the `content` member using the function, because the function has no implementation in the base class Message. So the compiler will complain about `content` needing to have a non-null value when exiting constructor.
I think having ParseRawMessage() ultimately return a string that we assign `content` to might be cleaner?
In general I'm having some trouble understanding your model so I might be confusing somethings.

Reply:
Content shouldn't be set from ParseRawMessage, it should take in content and parse it shouldn't it? Not sure what you mean the constructor needs content to have a non-null value, content gets passed from the main constructor to the base constructor:

![image](https://github.com/Ldev-The-Remaster/ChattingApp/assets/60454980/e92fe483-790f-45fe-a929-bacbf69a58c2)